server/methods.js

```diff
@@ -179,6 +180,41 @@ Meteor.methods({
     }
     return Instances.findOne({ _id: table_id }).slug;
   },
+  editadv(instanceid, newThreshold, newMaxQuestion, newMaxResponse, newLength, newStale, newIsHidden, newSocial) {
+    if (this.userId) {
+      let keys;
+      const email = Meteor.users.findOne({ _id: this.userId }).emails[0].address;
+      const instance = Instances.findOne({
+        _id: instanceid,
+      });
+      if (email === instance.admin || instance.moderators.indexOf(email) !== -1) {
+        Instances.update({
+          _id: instanceid,
+        }, {
+          $set: {
+            threshold: newThreshold,
+            new_length: newLength,
+            stale_length: newStale,
+            max_question: newMaxQuestion,
+            max_response: newMaxResponse,
+            hidden: newIsHidden,
+            social: newSocial,
+          },
+        }, (error, count, status) => {
+          if (error) {
+            errorKey = error.invalidKeys[0].name;
+            throw new Meteor.Error(errorKey);
+          }
+        });
+        if (keys) {
+          return keys;
+        }
+        return true;
+      }
```

Comment:
Instead of nesting all of these conditionals, check out "guard statements." Basically, instead of waiting until the very end of the function to return true or false, invert the condition and return at the top of the function. It's hard to say in words, so maybe an example. Instead of
```javascript
if (this.userId) {
  if (otherThing) {
    return true;
  }
  return false;
}
```
```javascript
if (_.isUndefined(this.userId)) {
  return false;
}
if (otherThing) {
  return true;
}
```
And now all the nesting is gone.